test/framework/ownerreference_helpers.go

```diff
@@ -171,8 +171,8 @@ var CoreOwnerReferenceAssertion = map[string]func([]metav1.OwnerReference) error
 		return HasExactOwners(owners, machineDeploymentController)
 	},
 	machineKind: func(owners []metav1.OwnerReference) error {
-		// Machines must be owned and controlled by a MachineSet or a KubeadmControlPlane, depending on if this Machine is part of a ControlPlane or not.
-		return HasOneOfExactOwners(owners, []metav1.OwnerReference{machineSetController}, []metav1.OwnerReference{kubeadmControlPlaneController})
+		// Machines must be owned and controlled by a MachineSet, MachinePool, or a KubeadmControlPlane, depending on if this Machine is part of a Machine Deployment, MachinePool, or ControlPlane.
+		return HasOneOfExactOwners(owners, []metav1.OwnerReference{machineSetController}, []metav1.OwnerReference{machinePoolController}, []metav1.OwnerReference{kubeadmControlPlaneController})
```

Comment:
Was there a test failing to prompt adding this? AFAICT `MachinePool` `Machines` aren't being deployed in any test at this point - but maybe I'm doing something wrong?

I'm expecting this to be part of the ClusterClass quickstart test.

Reply:
Let's revert this change here if it's not actually being tested in this PR.

Reply:
Just to confirm, are you saying this PR should be everything not in `test/` and then the `test/framework` change should go in #8842?

If so, makes sense to me 👍

Reply:
Yeah - let's move this change to #8842 so it's there with the actual test implementation.

Reply:
Still mixed about keeping this test case without testing it, but fine to keep it in on balance as it's not going to break anything that works currently, and could help existing users.